taichi/transforms/simplify.cpp

```diff
@@ -566,7 +566,7 @@ class BasicBlockSimplify : public IRVisitor {
   void visit(UnaryOpStmt *stmt) override {
     if (is_done(stmt))
       return;
-    if (stmt->op_type == UnaryOpType::cast) {
+    if (stmt->is_cast()) {
       if (stmt->cast_type == stmt->operand->ret_type.data_type) {
```

Comment:
Is this a pot-bug again? We didn't check `cast_by_value` here?

Reply:
It seems so.